### PR TITLE
migrations written

### LINKED ISSUE #6: Create migration scripts
These are knex migrations to create the tables. Dependent on #5.

- [x] `npm i knex sqlite3 --save`
- [x] Add a `knex` script to the `package.json`
- [x] `npm run knex migrate:make [table_name]`
- [x] Edit the migration file
- [x] `npm run knex migrate:latest`
- [x] Use a DB tool to ensure tables are created

Rinse and repeat 😉

### PR DESCRIPTION
**Pull request by JaivePNG**
Closes #6

### CODE REVIEW

**Comment by JaivePNG** [comment]
Conflicts resolved.

File: migrations/20161013143349_Projects.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      table.string('url')
+      table.date('date')
+    }),
+    knex.schema.createTable('photos', function(table){
+      table.increments('id').primary()
+      table.string('url')
+      table.string('caption')
+      table.integer('projects_id').references('projects.id')
+    }),
+    knex.schema.createTable('teammember', function(table){
```
Comment (locksmithdon) [inline migrations/20161013143349_Projects.js]: Please rename table to `team_members`. Don't forget to make the same change in the `dropTable` calls below.

File: migrations/20161013143349_Projects.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+
```
Comment (locksmithdon) [inline migrations/20161013143349_Projects.js]: Remove empty line :smile:

File: migrations/20161013143349_Projects.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      table.increments('id').primary()
+      table.string('name')
+    }),
+    knex.schema.createTable('projectteam', function(table){
+      table.increments('id').primary()
+      table.integer('teammember_id').references('teammember.id')
+      table.integer('project_id').references('projects.id')
+    })
+  ])
+
```
Comment (locksmithdon) [inline migrations/20161013143349_Projects.js]: And this one :smile_cat:

File: public/add.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
```
Comment (locksmithdon) [inline public/add.html]: Should this be part of this PR?

File: migrations/20161013143349_Projects.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      table.increments('id').primary()
+      table.string('url')
+      table.string('caption')
+      table.integer('projects_id').references('projects.id')
+    }),
+    knex.schema.createTable('teammember', function(table){
+      table.increments('id').primary()
+      table.string('name')
+    }),
+    knex.schema.createTable('projectteam', function(table){
```
Comment (locksmithdon) [inline migrations/20161013143349_Projects.js]: Please rename table to `project_teams`. Don't forget to make the same change in the `dropTable` calls below.

File: migrations/20161013143349_Projects.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      table.increments('id').primary()
+      table.integer('teammember_id').references('teammember.id')
```
Comment (locksmithdon) [inline migrations/20161013143349_Projects.js]: `team_member_id` and `team_member.id`

File: migrations/20161013143349_Projects.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+
+exports.up = function(knex, Promise) {
+  return Promise.all([
+    knex.schema.createTable('projects', function(table){
+      table.increments('id').primary()
+      table.string('title')
+      table.string('description')
+      table.string('url')
```
Comment (locksmithdon) [inline migrations/20161013143349_Projects.js]: Let's rename this to `repo_url` and add a new field called `app_url`.

**Comment by JaivePNG** [comment]
Changes made.

File: migrations/20161013143349-projects.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+      table.integer('projects_id').references('projects.id')
+    }),
+    knex.schema.createTable('team_members', function (table) {
+      table.increments('id').primary()
+      table.string('name')
+    }),
+    knex.schema.createTable('project_teams', function (table) {
+      table.increments('id').primary()
+      table.integer('team_members_id').references('team_members.id')
+      table.integer('projects_id').references('projects.id')
```
Comment (locksmithdon) [inline migrations/20161013143349-projects.js]: Darn, sorry. I missed these names during my code review. These foreign keys should be renamed to `team_member_id` and `project_id`. Can you please issue another PR?

File: migrations/20161013143349-projects.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+      table.string('description')
+      table.string('repo_url')
+      table.string('app_url')
+      table.date('date')
+    }),
+    knex.schema.createTable('photos', function (table) {
+      table.increments('id').primary()
+      table.string('url')
+      table.string('caption')
+      table.integer('projects_id').references('projects.id')
```
Comment (locksmithdon) [inline migrations/20161013143349-projects.js]: Sorry I didn't catch this during my code review, but this needs to be renamed to `project_id`.

**Comment by Andrew-Hird** [comment]
foreign keys properly named #43

**Pull request merged**